lib/src/source/hosted.dart

```diff
@@ -1100,3 +1109,91 @@ class _RefAndCache {
   @override
   bool operator ==(Object other) => other is _RefAndCache && other.ref == ref;
 }
+
+const checksumHeaderName = 'x-goog-hash';
+
+bool _hasChecksumHeader(Map<String, String> headers) {
```

Comment:
Perhaps just inline this helper

Reply:
Do you mean like this?

```dart
bool _hasChecksumHeader(Map<String, String> headers) =>
    headers.containsKey(checksumHeaderName);
```

Or remove the function and replace the function call?

```dart
var responseStream = response.headers.containsKey(checksumHeaderName)
          ? _validateChecksum(response, fileName)
          : response.stream;
```